pkg/api/helper/helpers.go

```diff
@@ -386,6 +386,40 @@ func NodeSelectorRequirementsAsSelector(nsm []api.NodeSelectorRequirement) (labe
 	return selector, nil
 }
 
+// NodeSelectorRequirementsAsSelectorWithougValidation converts the []NodeSelectorRequirement api type into a struct that implements
+// labels.Selector; it will NOT validate the key & values in the Requirement to improve performance.
+func NodeSelectorRequirementsAsSelectorWithoutValidation(nsm []api.NodeSelectorRequirement) (labels.Selector, error) {
```

Comment:
Can you factor out the common part of NodeSelectorRequirementsAsSelector and this function?